libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -191,6 +186,8 @@ constructor(
                 navigationOptions
             )
         }
+
+        fasterRouteController = FasterRouteController(directionsSession, tripSession)
```

Comment:
[NavigationComponentProvider](https://github.com/mapbox/mapbox-navigation-android/blob/2a08512a2c7db60609cf6b5672a39a223defca31/libnavigation-core/src/main/java/com/mapbox/navigation/core/NavigationComponentProvider.kt#L16-L44) is really a poltergeist class. 

so initializing this class directly here. though, It would be nice to have an object graph that kept track of dependencies. 

I saw that 1tap is using [Koin](https://github.com/InsertKoinIO/koin). I'm used to using Dagger1/Dagger2 (even though i don't really recommend them =) ).

Understanding that incorporating new frameworks into SDKs is a heated topic. Should we try to build our own lightweight dependency injection? Or could we use Koin? cc @LukasPaczos

Reply:
We've discussed this many times before 😅 And yeah it's indeed a heated topic 😂 In any case, let's discuss internally, this shouldn't be a blocker here (actually it's unrelated?). Thanks though for bringing that up though 🙇